Extensions/xkit_patches.js

```diff
@@ -124,33 +124,74 @@ XKit.tools.parse_version = function(versionString) {
 	return version;
 };
 
+var blogs_get = document.createElement("script");
+blogs_get.textContent = "("+(function(){
+	var blogs = [];
+	var models = Tumblr.dashboardControls.allTumblelogs;
+
+	models.filter(function(model){
+		return model.attributes.hasOwnProperty("is_current");
+	}).forEach(function(model){
+		blogs.push(model.attributes.name);
+	});
+	if(blogs.length){
+		window.postMessage({
+			xkit_blogs: blogs,
+		}, window.location.protocol+"//"+window.location.host);
+	}
+}).toString()+")();";
+document.body.appendChild(blogs_get);
+
+window.addEventListener("message", function(e){
+	if (e.origin == window.location.protocol+"//"+window.location.host &&
+		e.data.hasOwnProperty("xkit_blogs")) {
+
+		XKit.blogs_from_tumblr = e.data.xkit_blogs.map(XKit.tools.escape_html);
```

Comment:
because its a postmessage, so we don't really have any way of ensuring who's sending it. TBF its kinda duplicative with the origin check, but better safe then sorry.